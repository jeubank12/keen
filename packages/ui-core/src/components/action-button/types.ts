export type ButtonAction = 'create' | 'remove';
