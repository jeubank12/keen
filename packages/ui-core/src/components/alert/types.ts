export type Variant = 'success' | 'error';
