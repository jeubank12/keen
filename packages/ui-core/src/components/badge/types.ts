import { colors } from '@keen.io/colors';

export type Variant = keyof typeof colors;
