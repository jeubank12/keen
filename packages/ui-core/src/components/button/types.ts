export type ButtonVariant = 'primary' | 'secondary' | 'success' | 'danger';

export type ButtonSize = 'default' | 'large';

export type ButtonType = 'solid' | 'outline';
