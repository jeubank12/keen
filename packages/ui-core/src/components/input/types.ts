export type InputVariant = 'solid' | 'outline';
