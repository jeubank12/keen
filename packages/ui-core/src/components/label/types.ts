export type LabelVariant = 'primary' | 'secondary';
