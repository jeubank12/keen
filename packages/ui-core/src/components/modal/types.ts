export type ClosableIndicator = (isClosable: boolean) => void;
