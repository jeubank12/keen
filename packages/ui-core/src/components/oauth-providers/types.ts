export type OAuthConfig = {
  googleOAuth: {
    label: string;
    url: string;
    clientId: string;
    redirectUri: string;
    scope: string;
  };
  gitHubOAuth: {
    label: string;
    url: string;
    scope: string;
    clientId: string;
  };
};

export enum OAuthUserAction {
  REGISTER = 'signup',
  LOGIN = 'login',
  ACCEPT_INVITE = 'invite',
}

export type OAuthState = {
  /* OAuth flow action */
  action: OAuthUserAction;
  /* The URL of OAuth flow initiator */
  requestInitiatorUrl: string;
};
