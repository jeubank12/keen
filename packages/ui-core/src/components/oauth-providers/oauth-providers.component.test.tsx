import React from 'react';
import { render as rtlRender, fireEvent } from '@testing-library/react';

import OAuthProviders from './oauth-providers.component';
import { OAuthSource } from './types';

const render = (overProps: any = {}) => {
  const googleOAuth = {
    label: 'Sign up with Google',
    url: 'http://google',
    clientId: 'keen-client-id',
    redirectUri: 'keen-redirect',
    scope: 'email profile',
  };

  const gitHubOAuth = {
    label: 'Sign up with Github',
    url: 'https://github.com/login/oauth/authorize',
    scope: 'user:email',
    clientId: 'keen-client-id',
  };

  const host = 'https//keen-test.io';

  const props = {
    config: {
      googleOAuth,
      gitHubOAuth,
      host,
    },
    ...overProps,
  };

  const wrapper = rtlRender(<OAuthProviders {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<OAuthProviders />', () => {
  test('should redirect to Google OAuth page', () => {
    const {
      wrapper: { getByTestId },
    } = render({ source: OAuthSource.ACCEPT_INVITE });
    Object.defineProperty(window, 'location', {
      writable: true,
      value: {
        replace: jest.fn(),
      },
    });

    const button = getByTestId('google-oauth');
    fireEvent.click(button);

    const callArguments = window.location.replace['mock'].calls[0][0];

    expect(callArguments).toMatchSnapshot();
    expect(window.location.replace).toHaveBeenCalledWith(callArguments);
  });

  test('should redirect to Github OAuth page', () => {
    const {
      wrapper: { getByTestId },
    } = render({ source: OAuthSource.ACCEPT_INVITE });
    Object.defineProperty(window, 'location', {
      writable: true,
      value: {
        replace: jest.fn(),
      },
    });

    const button = getByTestId('github-oauth');
    fireEvent.click(button);

    const callArguments = window.location.replace['mock'].calls[0][0];

    expect(callArguments).toMatchSnapshot();
    expect(window.location.replace).toHaveBeenCalledWith(callArguments);
  });
});
