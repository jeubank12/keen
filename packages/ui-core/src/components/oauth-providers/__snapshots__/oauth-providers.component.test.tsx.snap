// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OAuthProviders /> should redirect to Github OAuth page 1`] = `"https://github.com/login/oauth/authorize?scope=user%3Aemail&client_id=keen-client-id&state=invite"`;

exports[`<OAuthProviders /> should redirect to Google OAuth page 1`] = `"http://google?scope=email%20profile&include_granted_scopes=true&response_type=code&redirect_uri=https%2F%2Fkeen-test.io%2Fkeen-redirect%3Fsource%3Dinvite&client_id=keen-client-id"`;
